app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/DisplayProfile.kt

```diff
@@ -231,6 +279,127 @@ fun ProfileHeader(
+                                    showReportOptions = false
+                                    selectedReportType = null
+                                  }) {
+                                    Text("Cancel")
+                                  }
+                              TextButton(
+                                  onClick = {
+                                    blockReportModal = false
+                                    showReportOptions = false
+                                    Toast.makeText(
+                                            context,
+                                            context.getString(
+                                                R.string.report_success_message,
+                                                selectedReportType?.displayName),
+                                            Toast.LENGTH_SHORT)
+                                        .show()
+                                    selectedReportType = null
+                                  },
+                                  enabled = selectedReportType != null) {
+                                    Text("Report")
+                                  }
+                            }
+                      } else if (showBlockConfirmation) {
+                        Row(
+                            modifier = Modifier.fillMaxWidth(),
+                            horizontalArrangement = Arrangement.SpaceBetween,
+                            verticalAlignment = Alignment.CenterVertically) {
+                              TextButton(
+                                  onClick = {
+                                    showBlockConfirmation = false
+                                    blockReportModal = false
+                                  }) {
+                                    Text("Cancel")
+                                  }
+                              TextButton(
+                                  onClick = {
+                                    showBlockConfirmation = false
+                                    blockReportModal = false
+                                    Toast.makeText(
+                                            context,
+                                            context.getString(R.string.block_success_message),
+                                            Toast.LENGTH_SHORT)
+                                        .show()
+                                  }) {
+                                    Text("Block")
+                                  }
+                            }
+                      } else {
+                        Row(
+                            modifier = Modifier.fillMaxWidth(),
+                            horizontalArrangement = Arrangement.SpaceBetween,
+                            verticalAlignment = Alignment.CenterVertically) {
+                              TextButton(onClick = { blockReportModal = false }) { Text("Cancel") }
```

Comment:
Nitpick: as you reuse the cancel string here, it would make even more sense to create a general one that you call.